instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriver.java

```diff
@@ -0,0 +1,230 @@
+   * the class is loaded. To avoid leaks, this method allow unregistering the driver so that the
+   * class can be gc'ed if necessary.
+   *
+   * @throws IllegalStateException if the driver is not registered
+   * @throws SQLException if deregistering the driver fails
+   */
+  public static void deregister() throws SQLException {
+    if (!REGISTERED.compareAndSet(true, false)) {
+      throw new IllegalStateException(
+          "Driver is not registered (or it has not been registered using Driver.register() method)");
+    }
+    DriverManager.deregisterDriver(INSTANCE);
+  }
+
+  /** Returns {@code true} if the driver is registered against {@link DriverManager}. */
+  public static boolean isRegistered() {
+    return REGISTERED.get();
+  }
+
+  private static Driver findDriver(String realUrl) {
+    for (Driver candidate : Collections.list(DriverManager.getDrivers())) {
+      try {
+        if (!(candidate instanceof OpenTelemetryDriver) && candidate.acceptsURL(realUrl)) {
+          return candidate;
+        }
+      } catch (SQLException ignored) {
+        // intentionally ignore exception
+      }
+    }
+
+    throw new IllegalStateException("Unable to find a driver that accepts url: " + realUrl);
+  }
+
+  /**
+   * Parses out the real JDBC connection URL by removing "otel:" prefix.
+   *
+   * @param url the connection URL
+   * @return the parsed URL
+   */
+  private static String extractRealUrl(String url) {
+    return url.startsWith(INTERCEPTOR_MODE_URL_PREFIX)
+        ? url.replace(INTERCEPTOR_MODE_URL_PREFIX, "jdbc:")
+        : url;
+  }
+
+  @Nullable
+  @Override
+  public Connection connect(String url, Properties info) throws SQLException {
+    if (url == null || url.trim().length() == 0) {
+      throw new IllegalArgumentException("url is required");
+    }
+
+    if (!acceptsURL(url)) {
+      return null;
+    }
+
+    final String realUrl = extractRealUrl(url);
+
+    // find the real driver for the URL
+    final Driver wrappedDriver = findDriver(realUrl);
+
+    final Connection connection = wrappedDriver.connect(realUrl, info);
+
+    final DbInfo dbInfo = JdbcConnectionUrlParser.parse(realUrl, info);
+
+    return new OpenTelemetryConnection(connection, dbInfo);
+  }
+
+  @Override
+  public boolean acceptsURL(String url) {
+    if (url == null) {
+      return false;
+    }
+    if (url.startsWith(INTERCEPTOR_MODE_URL_PREFIX)) {
+      return true;
+    }
+    return interceptorMode && url.startsWith("jdbc:") && url.length() > 5;
+  }
+
+  @Override
+  public DriverPropertyInfo[] getPropertyInfo(String url, Properties info) throws SQLException {
+    if (url == null || url.trim().length() == 0) {
+      throw new IllegalArgumentException("url is required");
+    }
+
+    final String realUrl = extractRealUrl(url);
+    final Driver wrappedDriver = findDriver(realUrl);
+    return wrappedDriver.getPropertyInfo(realUrl, info);
+  }
+
+  @Override
+  public int getMajorVersion() {
```

Comment:
I guess these are supposed to match the version we released. Can you parse these out of `InstrumentationVersion.VERSION`?

Reply:
Yeah, of course.

Reply:
Done.